### PR TITLE
Implement and demonstrate a sampling LOS

### PR DESCRIPTION
**Pull request by Edenhofer**

### CODE REVIEW

File: demos/0_intro.py [hunk 1]
```diff
@@ -187,17 +184,16 @@ def __call__(self, x):
...
 
-key, subkey = random.split(key)
-pos_init = jft.Vector(jft.random_like(subkey, signal_response.domain))
+key, k_i, k_o = random.split(key, 3)
 # NOTE, changing the number of samples always triggers a resampling even if
 # `resamples=False`, as more samples have to be drawn that did not exist before.
 samples, state = jft.optimize_kl(
     nll,
-    pos_init,
+    jft.Vector(jft.random_like(k_i, signal_response.domain)),
```
Comment (ph-frank) [inline demos/0_intro.py:192]: Shall we advertise `signal_response.init` here instead?

File: src/re/extra/sampling_los.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        end :
+            Location of the end point(s) in Cartesian space of shape
+            `(n_points, n_dim)`.
+        shape :
+            Shape of the input.
+        distances :
+            Tuple of distances for each axis of the shape of the input
+        n_sampling_points :
+            Number of sampling points per LOS for the integration.
+        dtype :
```
Comment (ph-frank) [inline src/re/extra/sampling_los.py:65]: Move docstring below `interpolation_order` and indicate that its optional.

File: src/re/extra/sampling_los.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+            distances=self.distances,
+            shape=jnp.array(shape),
+        )
+        super().__init__(
+            domain=ShapeWithDtype(shape, dtype), target=ShapeWithDtype(end.shape, dtype)
+        )
+
+    def __call__(self, x):
+        in_axes = (None, 0, 0)
+        if self.start.ndim < self.end.ndim:
```
Comment (ph-frank) [inline src/re/extra/sampling_los.py:87]: Contrary to the docstring this also allows for input of the form (ndim,) for `start` and `end`. Could this be emphasized in the doc?

**Review by ph-frank — APPROVED** [review]
Looks good! I only have tiny comments, after them its good to go!

File: demos/0_intro.py (hunk 1 above)
Comment (Edenhofer) [reply demos/0_intro.py:192]: yes, we absolutely should! Thanks! :)

**Comment by Edenhofer** [comment]
I incorporated all of them! Thanks for the review! :)

**Pull request merged**